### PR TITLE
updates for Julia 0.5

### PR DESCRIPTION
**Pull request by CarloLucibello**
Probably we will have to checkout some dependences to make this work

### CODE REVIEW

**Comment by CarloLucibello** [comment]
I'm really bewildered, I get an error in the community detection test here that I don't get on my machine. Will have to further investigate

**Comment by tbreloff** [comment]
Just curious what the status is here.  I uninstalled LightGraphs because it either downgrades other packages, or when I checkout:

```
WARNING: julia is fixed at 0.5.0-rc1+2 conflicting with requirement for LightGraphs: [0.4.0,0.5.0-)
```

**Comment by sbromberger** [comment]
What packages is it downgrading?

**Comment by tbreloff** [comment]
Freeing after having master checked out:

```
julia> Pkg.free("LightGraphs")
INFO: Freeing LightGraphs
INFO: Installing ArrayViews v0.6.4
INFO: Installing Clustering v0.5.0
INFO: Installing Combinatorics v0.3.2
INFO: Installing Distances v0.3.2
INFO: Downgrading LightGraphs: v0.6.1 => v0.5.4
INFO: Installing Polynomials v0.0.5
INFO: Downgrading StatsBase: v0.9.0 => v0.8.3
```

> we make no guarantees of compatibility with non-release versions of Julia

That's fine.  That's why I posted in this issue.  I just wanted to get a feel for the progress.

**Comment by sbromberger** [comment]
Ah - I may see the problem here. The issue is that we just put in the upper limit of `0.5-` for Julia; in v0.5.4 of LightGraphs we had no restriction so you're being forced to use that. If you get rid of the upper limit in REQUIRE, you might have better luck - we've done a TON of work since LG v0.5.4.

File: doc/build.jl [hunk 1]
```diff
@@ -133,7 +133,7 @@ types:
 
 ## Neighbors and Degree
 
-{{degree, indegree, outdegree, Δ, δ, Δout, δout, δin, Δin, degree_histogram, density, neighbors,
```
Comment (sbromberger) [inline doc/build.jl]: Why are we getting rid of this function? (I use it frequently.)

File: src/LightGraphs.jl [hunk 2]
```diff
@@ -10,17 +10,17 @@ using Clustering: kmeans
...
             reverse, reverse!, blkdiag, getindex, setindex!, show, print, copy, in,
-            sum, size, sparse, eltype, length, ndims, issym, transpose,
-            ctranspose, join, start, next, done, eltype, get
+            sum, size, sparse, eltype, length, ndims, issymmetric, transpose,
+            ctranspose, join, start, next, done, eltype, get, length
 
 
 # core
-export SimpleGraph, Edge, Graph, DiGraph, vertices, edges, src, dst,
+export SimpleGraph, Edge, EdgeIter, Graph, DiGraph, vertices, edges, src, dst,
```
Comment (sbromberger) [inline src/LightGraphs.jl]: EdgeIter should remain unexported. It's an internal construct we don't want to expose.

File: src/community/detection.jl [hunk 3]
```diff
@@ -39,13 +39,9 @@ end
...
 
 return : a matrix ϕ where ϕ[:,i] are the coordinates for vertex i.
-
-Note does not explicitly construct the `non_backtracking_matrix`.
-See `Nonbacktracking` for details.
-
 """
 function nonbacktrack_embedding(g::Graph, k::Int)
-    B = Nonbacktracking(g)
+    B, edgmap = non_backtracking_matrix(g)
```
Comment (sbromberger) [inline src/community/detection.jl]: typo: `edgemap`?

File: src/operators.jl [hunk 4]
```diff
@@ -265,11 +265,11 @@ sum(g::SimpleGraph) = ne(g)
 """sparse(g) is the adjacency_matrix of g"""
 sparse(g::SimpleGraph) = adjacency_matrix(g)
 
-#arrayfunctions = (:eltype, :length, :ndims, :size, :strides, :issym)
+#arrayfunctions = (:eltype, :length, :ndims, :size, :strides, :issymmetric)
 eltype(g::SimpleGraph) = Float64
 length(g::SimpleGraph) = nv(g)*nv(g)
 ndims(g::SimpleGraph) = 2
-issym(g::SimpleGraph) = !is_directed(g)
+issymmetric(g::SimpleGraph) = !is_directed(g)
```
Comment (sbromberger) [inline src/operators.jl]: might want to deprecate `issym`.

File: src/spectral.jl [hunk 5]
```diff
@@ -151,7 +151,7 @@ function non_backtracking_matrix(g::SimpleGraph)
         end
     end
 
-    B = zeros(Float64, m, m)
+    B = spzeros(Float64, m, m)
```
Comment (sbromberger) [inline src/spectral.jl]: Does this have performance regressions?

File: test/community/detection.jl [hunk 6]
```diff
@@ -58,11 +58,12 @@ for k=2:5
     add_edge!(z, (k-1)*n, k*n)
     c = community_detection_nback(z, k)
     a = collect(n:n:k*n)
+    @show c[a]
+    @show unique(c[a])
```
Comment (sbromberger) [inline test/community/detection.jl]: Will this be removed?

File: test/runtests.jl [hunk 7]
```diff
@@ -52,7 +52,7 @@ tests = [
     "graphdigraph",
     "persistence",
     "distance",
-    "spectral",
+    "spectral",    # Excluded GraphMatrices tests for the moment
```
Comment (sbromberger) [inline test/runtests.jl]: GraphMatrices is now part of LGExtras, so this comment doesn't apply any more.

**Comment by CarloLucibello** [comment]
closing in favor of #413

**Pull request closed without merging**